version: "3.7"

services:
  unbound:
    user: '1000:1000'
    network_mode: "container:gluetun"
    image: "klutchell/unbound:latest"
    volumes:
      - ${APP_DATA_DIR}/data/unbound:/etc/unbound
  
  pihole:
    network_mode: "container:gluetun"
    image: pihole/pihole
    restart: on-failure
    ports:
      # - 53:53
      # - 53:53/udp
      - ${APP_PI_HOLE_PORT}:80
    volumes:
      - ${APP_DATA_DIR}/data/pihole:/etc/pihole/
      - ${APP_DATA_DIR}/data/dnsmasq:/etc/dnsmasq.d/
    environment:
      - VIRTUAL_HOST="pihole.${DOMAIN}"
      - WEBPASSWORD=${APP_PASSWORD}
      - PIHOLE_DNS=unbound
    depends_on:
      - unbound
    labels:
      traefik.enable: true
      traefik.http.routers.traefik.rule: Host(`pihole.${DOMAIN}`)
      traefik.http.services.traefik.loadbalancer.server.port: $APP_PI_HOLE_PORT

